lib/theme.php

```diff
@@ -0,0 +1,54 @@
+<?php
+/**
+ * Experimental theme config functionality.
+ *
+ * @package gutenberg
+ */
+
+/**
+ * Returns a full config or one of the sections containing the
+ * setup for the theme found in a file, or a void array if none found.
+ *
+ * @param string $config_path Path to the config file.
+ * @param string $section_name Optional name of the section to pick.
+ * @return array Full theme config or its subset.
+ */
+function gutenberg_experimental_get_theme_config( $config_path, $section_name = null ) {
+	$empty_config = array();
+	if ( ! file_exists( $config_path ) ) {
+		return $empty_config;
+	}
+	$theme_config = json_decode(
+		file_get_contents( $config_path ),
```

Comment:
I'm getting the impression from the rest of this function that this is intending to be very precise in its validation of the configuration file being usable. One thing that's missed is that `file_get_contents` [will generate a warning](https://www.php.net/manual/en/function.file-get-contents.php#refsect1-function.file-get-contents-errors) if the file doesn't exist. Would we want to [absorb](https://www.php.net/manual/en/language.operators.errorcontrol.php) the warning? Or let the warning be logged?

Reply:
> One thing that's missed is that file_get_contents will generate a warning if the file doesn't exist.

There is a check for the file's existence one line earlier.

> Would we want to absorb the warning? Or let the warning be logged?

In extended the existing logic a bit and I didn't think about warnings if the file couldn't be processed. It feels like the best idea would be to come up with a proper JSON schema validation for the file at some point. It will not only give proper feedback to the theme's consumer but it will also support other use cases like the ability to mark translatable fields for externals tools like WP-CLI (related discussion start here: https://github.com/wp-cli/i18n-command/pull/210#issuecomment-628545550.

Reply:
> > One thing that's missed is that file_get_contents will generate a warning if the file doesn't exist.
> 
> There is a check for the file's existence one line earlier.

Ah. In truth, I did miss it. There are apparently a couple other cases though:

>An E_WARNING level error is generated if filename cannot be found, maxlength is less than zero, or if seeking to the specified offset in the stream fails.

https://www.php.net/manual/en/function.file-get-contents.php#refsect1-function.file-get-contents-errors

Reply:
True, there is always something that might go wrong. If this is how WordPress core handles it, I'll update code tomorrow 👍

Reply:
I simplified handling and added `@` to silence potential warnings.

Reply:
Noting that late today, I stumbled by accident to this conversation:

https://make.wordpress.org/core/2020/03/20/updating-the-coding-standards-for-modern-php/#comment-38241

Which points to this guideline:

https://make.wordpress.org/core/handbook/best-practices/coding-standards/php/#error-control-operator

Even given the guidelines here, I think the usage should be appropriate, given that it's being acknowledged and the error case is handled by the subsequent condition.